libafl_targets/src/coverage.rs

```diff
@@ -53,18 +55,43 @@ pub fn autotokens() -> Result<Tokens, Error> {
 #[no_mangle]
 pub static mut __afl_map_size: usize = EDGES_MAP_SIZE;
 pub use __afl_map_size as EDGES_MAP_PTR_SIZE;
-use libafl::bolts::ownedref::OwnedSliceMut;
+use libafl::{bolts::ownedref::OwnedMutSlice, observers::StdMapObserver};
 
 /// Gets the edges map from the `EDGES_MAP_PTR` raw pointer.
+/// Assumes a `len` of `EDGES_MAP_PTR_SIZE`.
 ///
 /// # Safety
 ///
-/// This function will crash if `EDGES_MAP_PTR` is not a valid pointer.
-/// The `EDGES_MAP_PTR_SIZE` needs to be smaller than, or equal to the size of the map.
+/// This function will crash if `edges_map_mut_ptr` is not a valid pointer.
+/// The [`edges_max_num`] needs to be smaller than, or equal to the size of the map.
 #[must_use]
-pub unsafe fn edges_map_from_ptr<'a>() -> OwnedSliceMut<'a, u8> {
-    debug_assert!(!EDGES_MAP_PTR.is_null());
-    OwnedSliceMut::from_raw_parts_mut(EDGES_MAP_PTR, EDGES_MAP_PTR_SIZE)
+pub unsafe fn edges_map_mut_slice<'a>() -> OwnedMutSlice<'a, u8> {
+    OwnedMutSlice::from_raw_parts_mut(edges_map_mut_ptr(), edges_max_num())
+}
+
+/// Gets a new [`StdMapObserver`] from the current [`edges_map_mut_slice`].
+///
+/// # Safety
+/// This will dereference [`edges_map_mut_ptr`] and crash if it is not a valid address.
+pub unsafe fn std_edges_map_observer<'a, S>(name: S) -> StdMapObserver<'a, u8, false>
```

Comment:
@andreafioraldi can you double check if this whole looks correct? Thx

Reply:
Also, should this use `ConstMapObserver` on non-ptr maps?

Reply:
> Also, should this use ConstMapObserver on non-ptr maps?

No, slices are not const, const maps is when the size is set at compile time.

> @andreafioraldi can you double check if this whole looks correct? Thx

tomorrow, I took vacation today